src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden || victim instanceof Carnivore) {
```

Comment:
[CODE PERFOMANCE] - The current implementation of the `bite` method uses an `if` statement that checks if the victim is hidden or is an instance of `Carnivore`. However, the `instanceof` check will always return `false` when the victim is hidden because hidden is a property specific to `Herbivore`. To simplify the logic and improve performance, you can check if the victim is an instance of `Herbivore` and is not hidden in a single condition.

Reply:
done